GraphicsProgram/Form1.cs

```diff
@@ -43,63 +52,40 @@ private void exitToolStripMenuItem_Click(object sender, EventArgs e)
             Application.Exit();
         }
 
-
-        
-
         private void runbutton_Click(object sender, EventArgs e)
         {
             //Update X & Y coardinte text fields on run command
-            int updateX = penX;
-            int updateY = penY;
-
-            string stringX = updateX.ToString();
-            string stringY = updateY.ToString();
-
-            textBox3.Text = stringX;
-            textBox2.Text = stringY;
-
+            var penPosition = new PenPosition();
+            penPosition.X = penX;
+            penPosition.Y = penY;
 
+            textBox3.Text = penPosition.GetXAsString();
+            textBox2.Text = penPosition.GetYAsString();
+            
             //String array to split multi line text input
             string[] textBoxLines = userinput.Lines;
 
             string[] commands = new string[] { "repeat", "loop", "if" };
-
-
-
+                       
             //Input parsing, split multilines to single lines then split the single lines into an array
             foreach (string line in textBoxLines)
             {
                 var splitString = line.Split();
 
                 if (line.Contains("circle"))
```

Comment:
So if i add this:

_userOperationStrategies.Single(x => x.AppliesTo(OperationType.Loop, ShapeType.Circle)).DoDrawing(myPen, penPosition, g, line);

_userOperationStrategies.Single(x => x.AppliesTo(OperationType.Repeat, ShapeType.Circle)).DoDrawing(myPen, penPosition, g, line);

Will all of these be ran if the word circle is in the text box?

Reply:
Yeah exactly, might be best to split the line out, then find the keywords for shape and operation...

So this might become (where c is the position of "circle" in the array and i is the position of "if" in the array)

`var splitString = line.Split();

var circleString = splitString[x];
var operationString = splitString[i];

_userOperationStrategies.Single(x => x.AppliesTo(operationString, circleString)).DoDrawing(myPen, penPosition, g, line);`